File: app/src/main/java/vn/loitp/app/activity/customviews/watchwhilelayout/layout/WWLMusicControlsOverlay.java
package vn.loitp.app.activity.customviews.watchwhilelayout.layout;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import loitp.basemaster.R;

/**
 * Created by thangn on 2/26/17.
 */

public class WWLMusicControlsOverlay extends FrameLayout implements View.OnClickListener {
    private final RelativeLayout controlsLayout;
    private final ImageView collapseBtn;
    private final ViewGroup bottomEndContainer;
    private final ImageView fullscreenBtn;
    private Listener listener;
    private boolean isHided;

    public WWLMusicControlsOverlay(Context context) {
        super(context);

        setClipToPadding(false);

        LayoutInflater.from(context).inflate(R.layout.wwl_default_controls_overlay, this);
        this.controlsLayout = (RelativeLayout) findViewById(R.id.controls_layout);
        this.collapseBtn = (ImageView) findViewById(R.id.player_collapse_button);
        this.collapseBtn.setOnClickListener(this);
        this.bottomEndContainer = (ViewGroup) findViewById(R.id.bottom_end_container);
        this.fullscreenBtn = (ImageView) this.bottomEndContainer.findViewById(R.id.fullscreen_button);
        this.fullscreenBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (this.listener != null) {
            if (v == collapseBtn) {
                this.listener.CO_doCollapse();
            } else if (v == fullscreenBtn) {
                this.listener.CO_doFullscreen(!this.fullscreenBtn.isSelected());
            }
        }
    }

    public void showControls() {
        this.controlsLayout.setVisibility(VISIBLE);
        this.isHided = false;
    }

    public void hideControls() {
        this.controlsLayout.setVisibility(GONE);
        this.isHided = true;
    }

    public void setListener(Listener l) {
        this.listener = l;
    }

    public void switchFullscreen(boolean selected) {
        this.fullscreenBtn.setSelected(selected);
    }

    public void toggleControls() {
        if (this.isHided) {
            showControls();
        } else {
            hideControls();
        }
    }

    public interface Listener {
        void CO_doCollapse();

        void CO_doFullscreen(boolean selected);
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/wwlMusic/layout/WWLMusicControlsOverlay.kt
package vn.loitp.app.activity.customviews.wwlMusic.layout

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.RelativeLayout
import vn.loitp.app.R

class WWLMusicControlsOverlay(context: Context) : FrameLayout(context), View.OnClickListener {

    private val layoutControls: RelativeLayout
    private val btPlayerCollapse: ImageView
    private val layoutBottomEndContainer: ViewGroup
    private val btFullscreen: ImageView
    private var listener: Listener? = null
    private var isHide = false

    override fun onClick(v: View) {
        if (v === btPlayerCollapse) {
            listener?.doCollapse()
        } else if (v === btFullscreen) {
            listener?.doFullscreen(!btFullscreen.isSelected)
        }
    }

    @Suppress("unused")
    fun showControls() {
        layoutControls.visibility = VISIBLE
        isHide = false
    }

    fun hideControls() {
        layoutControls.visibility = GONE
        isHide = true
    }

    fun setListener(l: Listener?) {
        listener = l
    }

    fun switchFullscreen(selected: Boolean) {
        btFullscreen.isSelected = selected
    }

    fun toggleControls() {
        if (isHide) {
            showControls()
        } else {
            hideControls()
        }
    }

    interface Listener {
        fun doCollapse()
        fun doFullscreen(selected: Boolean)
    }

    init {
        clipToPadding = false
        LayoutInflater.from(context).inflate(R.layout.wwl_music_default_controls_overlay, this)
        layoutControls = findViewById(R.id.layoutControls)
        btPlayerCollapse = findViewById(R.id.btPlayerCollapse)
        btPlayerCollapse.setOnClickListener(this)
        layoutBottomEndContainer = findViewById(R.id.layoutBottomEndContainer)
        btFullscreen = layoutBottomEndContainer.findViewById(R.id.btFullscreen)
        btFullscreen.setOnClickListener(this)
    }
}
